packages/react-components/src/components/timegraph-output-component.tsx

```diff
@@ -288,6 +298,59 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
         }
     }
 
+    private findElement(payload: { [key: string]: string | { [key: string]: string }}): TimeGraphEntry | number | undefined {
+        const startTimestampString: string = payload['startTimestamp'].toString();
+        const found = this.state.timegraphTree.findIndex(e => e.start === BigInt(startTimestampString));
+        const timeFilter = this.state.timegraphTree.filter(e => e.start === BigInt(startTimestampString));
+
+        if (timeFilter.length === 1) {
+            // Found unique timestamp match
+            const row = this.state.dataRows.findIndex(e => e.id === timeFilter[0].id);
+            if (row) {
+                return row;
+            }
+        }
+        else if (found === -1 && typeof(payload.load) !== 'string') {
+            // Parse identifiers from "Contents" field
+            const contents = payload.load.Contents.split(', ');
```

Comment:
So, your trying to match the Content column with the metadata. However, the idea of the this feature was to match the metadata with the columns of the event in the table, where the key is the header name and value the content of the cell. So, what you need to do is find the most intersecting row, i.e. most number of matching key/value pairs from metadata and event header/cell pairs. To test I tried to implement it with the following code:
```javascript
    private doHandleSelectionChangedSignal(payload: { [key: string]: string }) {
        const offset = this.props.viewRange.getOffset() || BigInt(0);
        const startTimestamp = payload['startTimestamp'];
        const endTimestamp = payload['endTimestamp'];
        if (startTimestamp !== undefined && endTimestamp !== undefined) {
            const selectionRangeStart = BigInt(startTimestamp) - offset;
            const selectionRangeEnd = BigInt(endTimestamp) - offset;
            const foundElement = this.findElement(payload);
            // Scroll vertically
            if (foundElement) {
                const rowIndex = this.state.dataRows.findIndex(d => d.id === foundElement.id);
                if (this.timeGraphTreeRef.current) {
                    this.timeGraphTreeRef.current.scrollTop = 20 * rowIndex;
                }
                // Select row
                const row = this.state.dataRows.find(d => d.id === foundElement.id);
                if (row) {
                    this.chartLayer.selectRow(row);
                    // TODO select state at row/starttime
                }
            }
            // Scroll horizontally
            this.props.unitController.selectionRange = {
                start: selectionRangeStart,
                end: selectionRangeEnd
            };
            this.chartCursors.maybeCenterCursor();
        }
    }

    private findElement(payload: { [key: string]: string | { [key: string]: string }}): TimeGraphEntry | undefined {
        let element: TimeGraphEntry | undefined = undefined;
        let max = 0;
        if (payload && payload.load) {
            this.state.timegraphTree.forEach(el => {
                if (el.metadata) {
                    let cnt = 0;
                    Object.entries(el.metadata).some(([key, value]) => {
                        if (typeof (payload.load) !== 'string') {
                            const val = payload.load[key];
                            if (val !== undefined) {
                                const num = Number(val);
                                if ((num !== undefined && num === value[0]) || (val === value[0])) {
                                    cnt++;
                                }
                            }
                        }
                    });
                    if (cnt > max) {
                        max = cnt;
                        element = el;
                    }
                }
            });
        }
        return element;
    }
```
The is probably a better way to implement the search algorithm `findElement` with some stream and lambdas and some javascript magic which you are more familiar that me. Feel free to reimplement the ```findElement``` method.

I tested my code with the `Thread Status` and `Resources` view.

Reply:
Thank you for the clarification and suggestions!
I tried a reducer in place of the `forEach` in `findElement`, but the solution was not much shorter than yours, just more abstract. So I preferred to kept your solution with a few minor adjustments, and cleaned up unused code.

Reply:
Thank you for the clarification and suggestions!
I tried a reducer in place of the `forEach` in `findElement`, but the solution was not much shorter than yours, just more abstract. So I preferred to kept your solution with a few minor adjustments, and cleaned up unused code.